### ISSUE TITLE
Win9.x emulation makes occasionally corrupted screens

### ISSUE THREAD

**Issue by tazok81** [joncampbell123/dosbox-x#4628]
### Describe the bug

I play Baldur's gate 1 in win98 emulation. From time to time I get broken screen. It happens randomly, sometimes it is rare, another time I get it 3-5 times over half hour. I experienced this issue in other games (polda 2, polda 3) and in win95 as well. I run it on windows 11 and tested it on ubuntu 22.04 too, both do the same.

In one second the screen messes :

![Screenshot from 2023-11-23 10-24-51](https://github.com/joncampbell123/dosbox-x/assets/146415614/28b86925-81af-40f6-bcba-0f67dd5c7f47)

in next second is fixed:

![Screenshot from 2023-11-23 10-25-57](https://github.com/joncampbell123/dosbox-x/assets/146415614/8ae15f70-d155-4571-8e18-82a39b6f80b3)

screen record:
https://github.com/joncampbell123/dosbox-x/assets/146415614/bdb6d526-1836-46b1-93af-429c50b8ec56

my config
[dosbox_x.conf.txt](https://github.com/joncampbell123/dosbox-x/files/13448119/dosbox_x.conf.txt)

### Steps to reproduce the behaviour

1. Start bg1 or any other win9.x game under win95/98. 
2. Play 5-20 minutes

### Expected behavior

Emulation will not produce corrupted frames

### What operating system(s) this bug have occurred on?

ubuntu 22.04, win11

### What version(s) of DOSBox-X have this bug?

2023.10.06

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Are these games using DirectX and hardware page flipping?

**Comment by tazok81** [comment]
Yes, all these games use DirectX. But I am not sure with second question. It is logical, but I dont know. Is there way how to check it? I can compile dosbox-x and add additional debug outputs if you need it.

**Comment by joncampbell123** [comment]
Try turning on the "video debug overlay" option in the Debug menu.
It will show stats around the DOS screen which includes the video memory offset and stride (bytes per line) below the emulator screen as well as the palette it uses. If that number often changes from zero to other values, page flipping is involved (or perhaps screen panning, but that's probably not involved here).

![image](https://github.com/joncampbell123/dosbox-x/assets/6245486/8d1fe8b5-9aec-4c13-ac9e-c27de1cbdf11)

**Comment by tazok81** [comment]
I turned on "Video debug overlay" in Help -> Logging console. This is what I see when I turn it on. Second image shows when it is turned off. I dont see such debug output like you have it. I see such overlay in dos games, but not in bg1.

![test1](https://github.com/joncampbell123/dosbox-x/assets/146415614/5fa4596d-6063-41a9-909c-4eafad869b33)
![test2](https://github.com/joncampbell123/dosbox-x/assets/146415614/1a8f4bb6-afa8-49fc-be78-1b23241838fb)

**Comment by tazok81** [comment]
Any idea how to catch some log when the mess happens? It's not often but I can invest the time to log it. Maybe, the simplest fix could to skip that broken frame.

**Comment by joncampbell123** [comment]
Set the output to Surface. I don't think I've tested that code against OpenGL or Direct3D mode.

**Comment by tazok81** [comment]
Thank you, will try it
